contracts/masset/Masset.sol

```diff
@@ -656,10 +832,48 @@ contract Masset is
         outputMinusFee = _bAssetQuantity.sub(feeAmount);
     }
 
+    /**
+     * vaultBalanceSum = totalSupply + 'surplus'
+     * maxCache = vaultBalanceSum * (cacheSize / 1e18)
+     * surplus is simply surplus, to reduce SLOADs
+     */
+    struct Cache {
+        uint256 vaultBalanceSum;
+        uint256 maxCache;
+        uint256 surplus;
+    }
+
+    /**
+     * @dev Gets the supply and cache details for the mAsset, taking into account the surplus
+     * @return Cache containing (tracked) sum of vault balances, ideal cache size and surplus
+     */
+    function _getCacheDetails() internal view returns (Cache memory) {
+        uint256 _surplus = surplus;
+        uint256 sum = totalSupply().add(_surplus);
+        return Cache(sum, sum.mulTruncate(cacheSize), _surplus);
+    }
+
     /***************************************
                     STATE
     ****************************************/
 
+    /**
+      * @dev Sets the MAX cache size for each bAsset. The cache will actually revolve around
+      *      _cacheSize * totalSupply / 2 under normal circumstances.
+      * @param _cacheSize Maximum percent of total mAsset supply to hold for each bAsset
+      */
+    function setCacheSize(uint256 _cacheSize)
+        external
+        onlyGovernance
```

Comment:
Other functions, such as `setSwapFee` are more restrictive in who can call them (i.e., `onlyGovernor`). Is there a specific reason for this?

Reply:
The mStable community DAO also has write access to some of the parameters, notably things like the cache size. These will be phased over gradually to onlyGovernance

Reply:
I see. I assumed something like this might be the reason, but wanted to make sure it wasn't a typo. :) Thanks for clarifying!